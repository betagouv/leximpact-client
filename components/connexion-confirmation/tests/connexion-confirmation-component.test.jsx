import { createShallow } from "@material-ui/core/test-utils";

import ConnexionConfirmationComponent from "../connexion-confirmation-component";

describe("components | connexion-confirmation-component", () => {
  let shallow;
  beforeAll(() => {
    const options = { dive: true };
    shallow = createShallow(options);
  });

  describe("snapshot", () => {
    it("doit correspondre quand le token est dans l'url", () => {
      const props = {
        handleUpdateConnexionToken: jest.fn(),
        onClosePopin: jest.fn(),
        token: "any-string",
      };
      const wrapper = shallow(<ConnexionConfirmationComponent {...props} />);
      expect(wrapper).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
