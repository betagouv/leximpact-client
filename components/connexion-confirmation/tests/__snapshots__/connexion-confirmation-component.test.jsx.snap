// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | connexion-confirmation-component snapshot doit correspondre quand le token est dans l'url 1`] = `
<div
  className="container"
>
  <WithStyles(IconButton)
    className="ConnexionConfirmationComponent-closeButton-1"
    onClick={[MockFunction]}
  >
    <pure(CloseIcon)
      fontSize="small"
    />
  </WithStyles(IconButton)>
  <h1
    className="ConnexionConfirmationComponent-h1Style-2"
  >
    <Icon
      icon={
        Object {
          "body": "<path fill=\\"#DD2E44\\" d=\\"M11.626 7.488a1.413 1.413 0 0 0-.268.395l-.008-.008L.134 33.141l.011.011c-.208.403.14 1.223.853 1.937.713.713 1.533 1.061 1.936.853l.01.01L28.21 24.735l-.008-.009c.147-.07.282-.155.395-.269 1.562-1.562-.971-6.627-5.656-11.313-4.687-4.686-9.752-7.218-11.315-5.656z\\"/><path fill=\\"#EA596E\\" d=\\"M13 12L.416 32.506l-.282.635.011.011c-.208.403.14 1.223.853 1.937.232.232.473.408.709.557L17 17l-4-5z\\"/><path fill=\\"#A0041E\\" d=\\"M23.012 13.066c4.67 4.672 7.263 9.652 5.789 11.124-1.473 1.474-6.453-1.118-11.126-5.788-4.671-4.672-7.263-9.654-5.79-11.127 1.474-1.473 6.454 1.119 11.127 5.791z\\"/><path fill=\\"#AA8DD8\\" d=\\"M18.59 13.609a.99.99 0 0 1-.734.215c-.868-.094-1.598-.396-2.109-.873-.541-.505-.808-1.183-.735-1.862.128-1.192 1.324-2.286 3.363-2.066.793.085 1.147-.17 1.159-.292.014-.121-.277-.446-1.07-.532-.868-.094-1.598-.396-2.11-.873-.541-.505-.809-1.183-.735-1.862.13-1.192 1.325-2.286 3.362-2.065.578.062.883-.057 1.012-.134.103-.063.144-.123.148-.158.012-.121-.275-.446-1.07-.532a.998.998 0 0 1-.886-1.102.997.997 0 0 1 1.101-.886c2.037.219 2.973 1.542 2.844 2.735-.13 1.194-1.325 2.286-3.364 2.067-.578-.063-.88.057-1.01.134-.103.062-.145.123-.149.157-.013.122.276.446 1.071.532 2.037.22 2.973 1.542 2.844 2.735-.129 1.192-1.324 2.286-3.362 2.065-.578-.062-.882.058-1.012.134-.104.064-.144.124-.148.158-.013.121.276.446 1.07.532a1 1 0 0 1 .52 1.773z\\"/><path fill=\\"#77B255\\" d=\\"M30.661 22.857c1.973-.557 3.334.323 3.658 1.478.324 1.154-.378 2.615-2.35 3.17-.77.216-1.001.584-.97.701.034.118.425.312 1.193.095 1.972-.555 3.333.325 3.657 1.479.326 1.155-.378 2.614-2.351 3.17-.769.216-1.001.585-.967.702.033.117.423.311 1.192.095a1 1 0 1 1 .54 1.925c-1.971.555-3.333-.323-3.659-1.479-.324-1.154.379-2.613 2.353-3.169.77-.217 1.001-.584.967-.702-.032-.117-.422-.312-1.19-.096-1.974.556-3.334-.322-3.659-1.479-.325-1.154.378-2.613 2.351-3.17.768-.215.999-.585.967-.701-.034-.118-.423-.312-1.192-.096a1 1 0 1 1-.54-1.923z\\"/><path fill=\\"#AA8DD8\\" d=\\"M23.001 20.16a1.001 1.001 0 0 1-.626-1.781c.218-.175 5.418-4.259 12.767-3.208a1 1 0 1 1-.283 1.979c-6.493-.922-11.187 2.754-11.233 2.791a.999.999 0 0 1-.625.219z\\"/><path fill=\\"#77B255\\" d=\\"M5.754 16a1 1 0 0 1-.958-1.287c1.133-3.773 2.16-9.794.898-11.364-.141-.178-.354-.353-.842-.316-.938.072-.849 2.051-.848 2.071a1 1 0 1 1-1.994.149c-.103-1.379.326-4.035 2.692-4.214 1.056-.08 1.933.287 2.552 1.057 2.371 2.951-.036 11.506-.542 13.192a1 1 0 0 1-.958.712z\\"/><circle fill=\\"#5C913B\\" cx=\\"25.5\\" cy=\\"9.5\\" r=\\"1.5\\"/><circle fill=\\"#9266CC\\" cx=\\"2\\" cy=\\"18\\" r=\\"2\\"/><circle fill=\\"#5C913B\\" cx=\\"32.5\\" cy=\\"19.5\\" r=\\"1.5\\"/><circle fill=\\"#5C913B\\" cx=\\"23.5\\" cy=\\"31.5\\" r=\\"1.5\\"/><circle fill=\\"#FFCC4D\\" cx=\\"28\\" cy=\\"4\\" r=\\"2\\"/><circle fill=\\"#FFCC4D\\" cx=\\"32.5\\" cy=\\"8.5\\" r=\\"1.5\\"/><circle fill=\\"#FFCC4D\\" cx=\\"29.5\\" cy=\\"12.5\\" r=\\"1.5\\"/><circle fill=\\"#FFCC4D\\" cx=\\"7.5\\" cy=\\"23.5\\" r=\\"1.5\\"/>",
          "height": 36,
          "width": 36,
        }
      }
      inline={true}
      width="40"
    />
     Vous êtes bien connectés à LexImpact POP 
  </h1>
  <p
    className="ConnexionConfirmationComponent-pStyle-3"
  >
    Vous pouvez désormais
    <b>
       simuler les impacts macros 
    </b>
    d’une réforme
    <br />
    sur la population 
    <Icon
      icon={
        Object {
          "body": "<path fill=\\"#FFAC33\\" d=\\"M11 0a5.964 5.964 0 0 0-3.402 1.061C7.401 1.031 7.205 1 7 1a4 4 0 0 0-4 4v6a7 7 0 1 0 14 0V6.001A6 6 0 0 0 11 0z\\"/><path fill=\\"#FFDC5D\\" d=\\"M11.213 34.309c-.052.682 1.849.953 2.129.791.299-.173-.698-.623-.702-.977-.019-1.797.53-1.648.311-3.492.443-1.625.672-2.592.672-3.131 0-1.25-3.312-.766-3.312 0 0 .344.365 1.681.417 3.131-.313 1.659.641 1.615.485 3.678zm-2.448 0c.052.682-1.849.953-2.129.791-.299-.173.698-.623.702-.977.019-1.797-.53-1.648-.311-3.492-.443-1.625-.672-2.592-.672-3.131 0-1.25 3.313-.766 3.313 0 0 .344-.365 1.681-.417 3.131.311 1.659-.642 1.615-.486 3.678z\\"/><path fill=\\"#F9CA55\\" d=\\"M10.311 27.5c0 .206.13.771.246 1.5h2.817c.164-.689.249-1.175.249-1.5 0-1.25-3.312-.766-3.312 0zm-3.957 0c0 .325.085.811.249 1.5H9.42c.116-.729.247-1.294.247-1.5 0-.766-3.313-1.25-3.313 0z\\"/><path fill=\\"#FFDC5D\\" d=\\"M3.258 26.96c-.258.124-.503-.097-.619-.227-.328-.367-.583-1.246-.282-2.209.303-.973.6-2.384 1.199-4.002.081-.22.499-3.764.805-4.445.197-.437 1.28-.896 1.81-.807.529.089.246.651.105 1.13 0 0-.916 3.953-1.016 4.296-.316 1.079-1.624 2.918-1.565 4.237.031.692.06 1.255-.339 1.246-.049.362-.003.735-.098.781zm14.984-.2c.271.091.487-.159.586-.302.279-.405.422-1.309.004-2.227-.423-.927-.894-2.29-1.69-3.821-.108-.208-.966-3.672-1.355-4.309-.25-.409-1.382-.729-1.897-.574-.513.154-.162.676.037 1.134 0 0 1.403 3.807 1.546 4.135.449 1.031 1.977 2.691 2.082 4.008.056.69.097 1.253.492 1.194.094.353.095.729.195.762z\\"/><path fill=\\"#292F33\\" d=\\"M13.945 36c.305 0 .372-.418.089-.629-.379-.285-.88-1.371-1.561-1.371h-1.245c-.229 0-.49 1.209-.41 1.48.102.345.433.52.812.52h2.315zm-7.913 0c-.305 0-.372-.418-.089-.629.379-.285.88-1.371 1.561-1.371h1.245c.229 0 .49 1.209.41 1.48-.102.345-.433.52-.811.52H6.032z\\"/><path fill=\\"#E75A70\\" d=\\"M14.354 17.771L16 16c-1.438-1.792-4-2-4-2H8s-2.562.208-4 2l1.646 1.771S6.25 20.688 6.25 21 5 25.75 5 28h10c0-2.25-1.25-6.688-1.25-7s.604-3.229.604-3.229z\\"/><path fill=\\"#DA2F47\\" d=\\"M13.75 21c0-.106.07-.512.161-1H6.089c.092.488.161.894.161 1 0 .083-.122.509-.223 1h7.942c-.108-.558-.219-.917-.219-1z\\"/><path fill=\\"#FFDC5D\\" d=\\"M8 11v4a2 2 0 0 0 4 0v-4H8z\\"/><path fill=\\"#F9CA55\\" d=\\"M12 11l-4 .019V15s2 1 4 0v-4z\\"/><path fill=\\"#FFDC5D\\" d=\\"M15 3H5v6c0 2.762 2.238 5 5 5s5-2.237 5-5V3z\\"/><path fill=\\"#FFAC33\\" d=\\"M15 2H5L4 7c4 0 4-4 4-4s0 4 8 4l-1-5z\\"/><path fill=\\"#DF1F32\\" d=\\"M12 12H8s0 1 2 1 2-1 2-1z\\"/><circle fill=\\"#662113\\" cx=\\"7.5\\" cy=\\"8.5\\" r=\\".5\\"/><circle fill=\\"#662113\\" cx=\\"12.5\\" cy=\\"8.5\\" r=\\".5\\"/><path fill=\\"#C1694F\\" d=\\"M9 10h2s0 1-1 1-1-1-1-1z\\"/><path fill=\\"#FFAC33\\" d=\\"M26.5 0a5.469 5.469 0 0 0-3.181 1.018A3.492 3.492 0 0 0 20 4.5c0 1.542.007 3.484 2.038 4.208A5.487 5.487 0 0 0 26.5 11a5.5 5.5 0 1 0 0-11z\\"/><path fill=\\"#FFDC5D\\" d=\\"M18.117 26.76c-.271.091-.487-.159-.586-.302-.279-.405-.422-1.309-.004-2.227.423-.927.702-2.072 1.69-3.821.115-.204.494-2.746 1.242-4.354.202-.435 1.495-.684 2.009-.529.513.154-.005.98-.037 1.134-.344 1.651-1.305 4.077-1.45 4.404-.8 1.806-1.965 2.389-2.071 3.706-.055.69-.205 1.286-.6 1.228-.092.352-.094.728-.193.761zm14.595.204c.257.126.504-.094.62-.222.33-.365.591-1.242.296-2.208-.297-.975-.424-2.146-1.174-4.01-.087-.217-.129-2.787-.66-4.479-.143-.457-1.392-.875-1.922-.789-.529.086-.124.972-.112 1.129.124 1.682.758 4.213.859 4.556.556 1.895 1.634 2.627 1.566 3.946-.036.692.034 1.302.433 1.296.048.363 0 .735.094.781z\\"/><path fill=\\"#AAB8C2\\" d=\\"M29.555 29.5c.5-2.734.422-6.5.422-6.5h-7.963s-.078 3.766.422 6.5c.063.344.188 3.344-.109 5.5-.066.479 2.886.937 2.844-.125-.078-1.969.264-3.513.328-4.5.035-.547.25-1.875.344-3.208.011-.018.295-.018.306 0 .094 1.333.308 2.661.344 3.208.064.987.406 2.531.328 4.5-.042 1.062 2.91.604 2.844.125-.297-2.156-.172-5.156-.11-5.5z\\"/><path fill=\\"#292F33\\" d=\\"M30.54 36c.499-.406-.041-1.161-1.42-1.402-1.163-.203-2.2-.032-2.363.35-.212.493-.05 1.052-.05 1.052h3.833zm-9.117 0c-.499-.406.041-1.161 1.42-1.402 1.163-.203 2.2-.032 2.363.35.212.494.05 1.053.05 1.053L21.423 36z\\"/><path fill=\\"#4289C1\\" d=\\"M30.797 14.766C29.578 14 28 14 28 14h-4s-1.578 0-2.797.766c-1.201.754-1.656 3-1.656 3l2.203.688.109 5.546h8.281l.109-5.547 2.203-.688c.001.001-.454-2.245-1.655-2.999z\\"/><path fill=\\"#FFDC5D\\" d=\\"M24 10.042v4c0 .682.896 1.305 2 1.305s2-.668 2-1.305v-4h-4z\\"/><path fill=\\"#F9CA55\\" d=\\"M24 14.042c0 .682.896 1.305 2 1.305s2-.668 2-1.305V11l-4 .019v3.023z\\"/><path fill=\\"#FFDC5D\\" d=\\"M31 3H21v6c0 2.762 2.238 5 5 5s5-2.237 5-5V3z\\"/><path fill=\\"#FFAC33\\" d=\\"M21 2.5L20.5 5l.5 2c2.333 0 2.85-2.412 2.965-3.488C24.004 4.588 24.779 7 31 7l.5-2-.5-2.5H21z\\"/><path fill=\\"#C1694F\\" d=\\"M28 12h-4s0 1 2 1 2-1 2-1z\\"/><circle fill=\\"#662113\\" cx=\\"23.5\\" cy=\\"8.5\\" r=\\".5\\"/><circle fill=\\"#662113\\" cx=\\"28.5\\" cy=\\"8.5\\" r=\\".5\\"/><path fill=\\"#C1694F\\" d=\\"M25 10h2s0 1-1 1-1-1-1-1z\\"/>",
          "height": 36,
          "width": 36,
        }
      }
      inline={true}
      width="25"
    />
     et les recettes de l’État 
    <Icon
      icon={
        Object {
          "body": "<path fill=\\"#292F33\\" d=\\"M7 11h22v18H7z\\"/><path fill=\\"#CCD6DD\\" d=\\"M8 29c0 1.104-.597 2-1.333 2H5.333C4.597 31 4 30.104 4 29V11c0-1.104.597-2 1.333-2h1.333C7.403 9 8 9.896 8 11v18zm24 0c0 1.104-.597 2-1.333 2h-1.334C28.597 31 28 30.104 28 29V11c0-1.104.597-2 1.333-2h1.334C31.403 9 32 9.896 32 11v18zm-16 0c0 1.104-.597 2-1.333 2h-1.333C12.597 31 12 30.104 12 29V11c0-1.104.597-2 1.333-2h1.333C15.403 9 16 9.896 16 11v18zm8 0c0 1.104-.598 2-1.334 2h-1.332C20.598 31 20 30.104 20 29V11c0-1.104.598-2 1.334-2h1.332C23.402 9 24 9.896 24 11v18z\\"/><path fill=\\"#9AAAB4\\" d=\\"M33 30a2 2 0 0 1-2 2H5a2 2 0 0 1 0-4h26a2 2 0 0 1 2 2z\\"/><path fill=\\"#CCD6DD\\" d=\\"M35 32a2 2 0 0 1-2 2H3a2 2 0 0 1 0-4h30a2 2 0 0 1 2 2z\\"/><path fill=\\"#E1E8ED\\" d=\\"M36 33.5a1.5 1.5 0 0 1-1.5 1.5h-33a1.5 1.5 0 1 1 0-3h33a1.5 1.5 0 0 1 1.5 1.5z\\"/><path fill=\\"#9AAAB4\\" d=\\"M33 10c0-1.104-.956-2-2.133-2H5c-1.179 0-2 .896-2 2 0 .751.386 1.398 1 1.74V13h4v-1h4v1h4v-1h4v1h4v-1h4v1h4v-1.312c.599-.354 1-.975 1-1.688z\\"/><path fill=\\"#CCD6DD\\" d=\\"M2 8.444C2 7.413 3.012 7 3.012 7l14.904-7 15.047 7S34 7.231 34 8.45V9H2v-.556z\\"/><path fill=\\"#9AAAB4\\" d=\\"M18 2.542S7.681 7.407 6.65 7.844C5.619 8.281 5.964 9 6.651 9h22.646c1.062 0 .812-.812-.031-1.25C28.422 7.312 18 2.542 18 2.542z\\"/><path fill=\\"#CCD6DD\\" d=\\"M34 9a1 1 0 0 1-1 1H3a1 1 0 0 1 0-2l30 .006c.553 0 1 .442 1 .994z\\"/>",
          "height": 36,
          "width": 36,
        }
      }
      inline={true}
      width="25"
    />
    <span>
      .
    </span>
  </p>
</div>
`;
